python/cudf/cudf/core/index.py

```diff
@@ -1929,7 +1929,40 @@ def floor(self, field):
         DatetimeIndex(['2020-05-31 08:59:00', '1999-12-31 18:44:00'],
         dtype='datetime64[ns]', freq=None)
         """
-        out_column = self._values.floor(field)
+        out_column = self._values.floor(freq)
+
+        return self.__class__._from_data({self.name: out_column})
+
+    def round(self, freq):
+        """
+        Perform round operation on the data to the specified freq.
+
+        Parameters
+        ----------
+        freq : str
+            One of ["D", "H", "T", "min", "S", "L", "ms", "U", "us", "N"].
+            Must be a fixed frequency like 'S' (second) not 'ME' (month end).
+            See `frequency aliases <https://pandas.pydata.org/docs/\
+                user_guide/timeseries.html#timeseries-offset-aliases>`__
+            for more details on these aliases.
+
+        Returns
+        -------
+        DatetimeIndex
+            Index of the same type for a DatetimeIndex
+
+        Examples
+        --------
+        >>> import cudf
+        >>> import pandas as pd
+        >>> rng = cudf.Index(pd.date_range('1/1/2018 11:59:00',
+        ... periods=3, freq='min'))
+        >>> rng.round('H')
+        DatetimeIndex(['2018-01-01 12:00:00', '2018-01-01 12:00:00',
+               '2018-01-01 12:00:00'],
+              dtype='datetime64[ns]', freq=None)
```

Comment:
It would be nice to have multiple output values that are not all identical. Maybe start at 11:59, with a frequency of days, and round to hours. Then all the 11:59s will round to 12:00 on different days? Or perhaps use an initial time of 11:59, frequency of 3 hours, and round to the nearest hour.

Also show the output of `>>> rng` before rounding, so that users can see the before/after more clearly.

You could adapt the examples from `Series.dt.round` if you want.